### DISCUSSION TITLE
IKEA Parasoll does not send battery status in real time

### DISCUSSION THREAD

**Discussion by Lacimaci78** [Koenkk/zigbee2mqtt#23156] [Devices]
IKEA Parasoll does not send battery status in real time. Sends the battery status from the first loading of the device to the Z2M. Even if I change and put a weaker battery in the device, this actual state will not be transferred to the Z2M

**Comment by Lacimaci78** [comment]
I used my own parasoll.js converter for the test, I set the time min: 60 and change: 1 the device sent its initial battery status after initial loading. Subsequently, I put the battery in the device at only 48%, this state was transferred to HA MQTT within a minute. So the battery is finally in good condition. And not just the initial one loaded during the first configuration of the device. and then never real, only primordial.

Here is the code:

const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const e = exposes.presets;

const definition = {
    zigbeeModel: ['PARASOLL Door/Window Sensor'],
    model: 'E2013',
    vendor: 'IKEA',
    description: 'PARASOLL Door/Window Sensor',
    fromZigbee: [fz.ias_contact_alarm_1, fz.battery],
    toZigbee: [],
    exposes: [e.battery(), e.contact()],
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg']);
        await reporting.batteryPercentageRemaining(endpoint, {min: 3600, max: 43200, change: 1});
        await endpoint.read('genPowerCfg', ['batteryPercentageRemaining']);
    },
};

module.exports = definition;

**Comment by lrgabriel** [comment]
I think I have the same problem.  All of my Parasolls have reported the same battery percentage since I initially installed them.

**Reply by Lacimaci78** [reply]
What batteries did you use in the magnets? and how long did they last?

**Reply by lrgabriel** [reply]
I don't have particularly helpful info here; I'm using LADDA 750 mAh AAAs.  I (shamefully) didn't fully charge the batteries before installing everything, so I started at ~80%.  It has been about a month, and the first (most used) sensor just died.

**Reply by sKinyo** [reply]
What do you mean with died? The battery or the device itself.

**Reply by lrgabriel** [reply]
The battery, sorry for lack of clarity.

**Comment by bertramdieterich** [comment]
Same here. I have approx. 30 IKEA Parasoll sensors installed, with a mix of different batteries (IKEA Ladda, Amazon Basics, a few Alkalines). Through the board, all sensors properly report opening/closing (so the network is fine). However, they all show the battery level from when they were paired (some not even that ... the "battery" field is just missing in the status and the UI shows "Battery, power level: OK" instead of a percentage).
To me it looks like zigbee2mqtt is unable (maybe just misconfigured?) to process the battery messages from the devices.

**Reply by turbolooser** [reply]
same issue here. battery stats not updating...

**Comment by rjblake** [comment]
Same here - does not update. I can manually update by doing the following:

1. From Z2M, select device (e.g. Front Door Sensor)
2. Select Exposes tab
3. Open or close door to wake sensor up
4. Click on the 'refresh' icon next to battery. Battery value gets updated

Based on this, it seems sensor only sends battery update info if polled and when the sensor is awake. It seems it does not send when the device sends the open/closed status. I see under the 'Bind' tab, that 'PowerCfg' is not bound, but I'm not sure if this would help. I've checked the bind status on an IKEA Styrbar remote and it is ticked there. Guess I'll give it a try to see if it does make a difference. Nothing to lose (famous last words)

**Comment by bertramdieterich** [comment]
Based on the discovery of rjblake I got it "kind of" working.

- Zigbee2Mqtt allows you to poll the battery level via MQTT by sending `{"battery":""}` to the topic `zigbee2mqtt/<device friendly name>/get`
- This only works, when the device is awake, so I used HA to poll this every time any PARASOL sensor sends "open"
- Now the sensor battery state nicely updates every time a window/door is opened
- The solution is still not perfect, because I use the sensors also in a few places, where they never get triggerd naturally (for alarm :-). So I have to remember to open those windows/doors at least 2x per year.

**Comment by turbolooser** [comment]
Battery reporting is still not working and then suddenly the sensor dies. Not sure we will ever get over this like here: https://github.com/Koenkk/zigbee2mqtt/issues/22579

**Comment by rjblake** [comment]
Not sure how/why, but seems my battery is being updated without any additional polling. Perhaps just ticking the PowerCfg under Bind tab does the trick

**Comment by evorgevol** [comment]
Hi, is there an update/workaround on this? My batteries are still always reporting 100%, and suddenly it stops working when the batteries die and I'm unaware. The payload always has battery 100:
```{"battery":100,"contact":false,"identify":null,"linkquality":104,"update":{"installed_version":16777241,"latest_version":16777241,"state":"idle"}}```

I do have the PowerCfg checked on the Bind tab, and have always had it. I have 6 of them and all behaving the same.

Thanks

**Reply by tkintscher** [reply]
Some time ago it started working for me:
<img width="1417" alt="Screenshot 2025-04-17 at 17 12 57" src="https://github.com/user-attachments/assets/07cd3a3c-eee4-4235-bd70-b3fe8e976df2" />

These are my bind settings:
<img width="1218" alt="Screenshot 2025-04-17 at 17 13 16" src="https://github.com/user-attachments/assets/4026a8da-34b9-4f75-905e-7b9055c27810" />

And these are my reporting settings:
<img width="1409" alt="Screenshot 2025-04-17 at 17 13 25" src="https://github.com/user-attachments/assets/a1439dc7-60d5-4adf-9d01-d24a401334a3" />

Hope that helps!

**Reply by evorgevol** [reply]
hi @tkintscher , thanks for your reply. I have since noticed that one of my six door sensors is working correctly.. but I have absolutely no idea why! The `Bind` and `Reporting` tabs each have the same (as does every tab). I've checked all other 5 and they all have the same, but battery % is still only being reported at 100%. I'll keep monitoring and playing to see if I can work it out!
